BaseClasses.py

```diff
@@ -754,20 +737,116 @@ def can_reach_entrance(self, spot: str, player: int) -> bool:
     def can_reach_region(self, spot: str, player: int) -> bool:
         return self.multiworld.get_region(spot, player).can_reach(self)
 
-    def sweep_for_events(self, locations: Optional[Iterable[Location]] = None) -> None:
+    def _sweep_for_events_impl(self, events_per_player: List[Tuple[int, List[Location]]], yield_each_sweep: bool,
+                               ) -> Iterable[None]:
```

Comment:
```suggestion
                               ) -> Iterator[None]:
```
The `Iterator` return type annotation is a common signal to people that this is an object that can only be iterated once - rather than some kind of collection, like a list or tuple, that can be iterated multiple times.

Reply:
Looking at the docs, `Iterable` requires that `__iter__()` is implemented whereas `Iterator` requires that both `__iter__()` and `__next__()` are implemented, so `Iterator` is more accurate too.

Before I commit the change, is there any preference or reasons against specifying the return type even more accurately as `Generator[None, None, None]` instead?

Reply:
I think I would find `Generator[None, None, None]` to be more noisy without much added value. But it's not bad if you really want it.